### DISCUSSION TITLE
Custom B2B prices - Is the only way to create a channel for each customer?

### DISCUSSION THREAD

**Discussion by V-iktor** [saleor/saleor#11624] [Q&A] [answered]
I was wondering how to implement custom prices for each customer.

Customer A gets a 5% discount on products X, Y, Z.

Customer B gets a 10% discount on products X, S, D.

Customer C gets not get discount on any items.

e.t.c.

My understanding is that the only way to implement this in Saleor is to create a separate channel for each customer, is that the case?

I would like to display the base price and the discounted price(if applicable) for every customer.

**Comment by timuric** [comment, marked as answer]
@V-iktor you can achieve that either with voucher, separate channel, or create custom price logic and override checkout prices https://docs.saleor.io/docs/3.x/developer/checkout#setting-custom-prices-for-checkout-items
